### PR TITLE
feat: [SIG-526]: UI Integrations V0

### PR DESCRIPTION
**Pull request by Vikrant2520**
### Summary

- Integrations V0 UI module 

#### Related Issues / PR's

https://linear.app/signoz-io/issue/SIG-526/ui-integrations-v0

#### Screenshots

https://github.com/SigNoz/signoz/assets/54737045/235eb658-28fa-464e-af68-6adf887caa69

### CODE REVIEW

**Comment by Vikrant2520** [comment]
Light Mode theme is pending...

**Comment by srikanthccv** [comment]
Please make this available under a hidden/private route. This will allow us to merge the changes to develop while we polish the work.

File: frontend/src/container/SideNav/menuItems.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {
 	ScrollText,
 	Settings,
 	Slack,
+	// Unplug,
```
Comment (YounixM) [inline frontend/src/container/SideNav/menuItems.tsx:19]: Remove commented code.

File: frontend/src/container/SideNav/menuItems.tsx [hunk 2]
```diff
@@ -89,6 +90,11 @@ const menuItems: SidebarItem[] = [
 		label: 'Alerts',
 		icon: <BellDot size={16} />,
 	},
+	// {
```
Comment (YounixM) [inline frontend/src/container/SideNav/menuItems.tsx:93]: Remove commented code.

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailContentTabs/IntegrationDetailContentTabs.styles.scss [hunk 3]
```diff
@@ -0,0 +1,296 @@
+.integration-detail-overview {
+	display: flex;
+
+	.integration-detail-overview-left-container {
+		display: flex;
+		flex-direction: column;
+		width: 25%;
+		gap: 26px;
+		border-right: 1px solid #161922;
```
Comment (YounixM) [inline frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailContentTabs/IntegrationDetailContentTabs.styles.scss:9]: use variable

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+					<TestConnection connectionState={connectionState} />
+					{connectionState === ConnectionStates.Connected ||
+					connectionState === ConnectionStates.NoDataSinceLong ? (
+						<>
+							<div className="data-info">
+								<Typography.Text className="last-data">
+									Last recieved from
+								</Typography.Text>
+								<Typography.Text className="last-value">
+									redis.service.alert
```
Comment (YounixM) [inline frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx:117]: Replace with actual value

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+								</Typography.Text>
+								<Typography.Text className="last-value">
+									27.02.2024⎯10:30:23
+								</Typography.Text>
+							</div>
+						</>
+					) : connectionState === ConnectionStates.TestingConnection ? (
+						<div className="data-test-connection">
+							<div className="last-data">
+								After adding the {title} integration, you need to manually configure
```
Comment (YounixM) [inline frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx:132]: move to strings

File: frontend/src/container/SideNav/menuItems.tsx (hunk 1 above)
Comment (Vikrant2520) [reply frontend/src/container/SideNav/menuItems.tsx:19]: This is required, we need to add back the integrations once the BE is ready.

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailPage.tsx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+				onClick={(): void => {
+					setSelectedIntegration(null);
+				}}
+			>
+				All Integrations
+			</Button>
+
+			{loading ? (
+				<div className="loading-integration-details">
+					Please wait.. While we load the integration details
```
Comment (YounixM) [inline frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailPage.tsx:96]: We should try and maintain standard format for loading and error states

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailPage.tsx [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+			) : isError ? (
+				<div className="error-container">
+					<div className="error-content">
+						<img
+							src="/Icons/awwSnap.svg"
+							alt="error-emoji"
+							className="error-state-svg"
+						/>
+						<Typography.Text>
+							Something went wrong :/ Refresh the page or contact support.
```
Comment (YounixM) [inline frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailPage.tsx:107]: Refresh should be triggered on the link click

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationsUninstallBar.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+	const handleCancel = (): void => {
+		setIsModalOpen(false);
+	};
+	return (
+		<div className="uninstall-integration-bar">
+			<div className="unintall-integration-bar-text">
+				<Typography.Text className="heading">Remove Integration</Typography.Text>
+				<Typography.Text className="subtitle">
+					Removing the {integrationTitle} integration would make your workspace stop
```
Comment (YounixM) [inline frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationsUninstallBar.tsx:56]: Move to strings

File: frontend/src/pages/Integrations/IntegrationsList.tsx [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+			return data?.data.data.integrations.filter((item) =>
+				item.title.toLowerCase().includes(searchTerm.toLowerCase()),
+			);
+		}
+		return [];
+	}, [data?.data.data.integrations, searchTerm]);
+
+	const loading = isLoading || isFetching || isRefetching;
+
+	const handleContactSupport = (): void => {
```
Comment (YounixM) [inline frontend/src/pages/Integrations/IntegrationsList.tsx:42]: Move this to a util function

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailPage.tsx (hunk 7 above)
Comment (Vikrant2520) [reply frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailPage.tsx]: We are retrying the API call that failed here, renamed it to `please retry`

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailContentTabs/IntegrationDetailContentTabs.styles.scss (hunk 3 above)
Comment (Vikrant2520) [reply frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailContentTabs/IntegrationDetailContentTabs.styles.scss]: Done

File: frontend/src/pages/Integrations/IntegrationsList.tsx (hunk 9 above)
Comment (Vikrant2520) [reply frontend/src/pages/Integrations/IntegrationsList.tsx]: done

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx (hunk 4 above)
Comment (Vikrant2520) [reply frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx]: done

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx (hunk 5 above)
Comment (Vikrant2520) [reply frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx:175]: will move in next PR

**Review by YounixM — APPROVED** [review]
LGTM

File: frontend/src/container/SideNav/menuItems.tsx (hunk 2 above)
Comment (Rajat-Dabade) [inline frontend/src/container/SideNav/menuItems.tsx:93]: Commented code.

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+	let latestData: {
+		last_received_ts_ms: number | null;
+		last_received_from: string | null;
+	} = {
+		last_received_ts_ms: null,
+		last_received_from: null,
+	};
+
+	if (
```
Comment (Rajat-Dabade) [inline frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx:75]: NIT: can we move this logic to utils

File: frontend/src/container/SideNav/menuItems.tsx (hunk 2 above)
Comment (Vikrant2520) [reply frontend/src/container/SideNav/menuItems.tsx:93]: we need this as we will uncomment it once BE is ready

File: frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx (hunk 10 above)
Comment (Vikrant2520) [reply frontend/src/pages/Integrations/IntegrationDetailPage/IntegrationDetailHeader.tsx:75]: yes will do

**Pull request merged**